core/types/src/macros.rs

```diff
@@ -68,6 +68,83 @@ macro_rules! impl_newtype {
     )*}
 }
 
+/// This macro provides common byte-handling operations when the type being
+/// wrapped is a struct containing a single fixed-size array of bytes.
+///
+/// This should be called from within a private submodule.
+/// TODO: Delete this once Display has been implemented for all callers.
+#[macro_export]
+macro_rules! impl_newtype_for_bytestruct_no_display {
+    ($($wrapper:ident, $inner:ident, $size:ident, $fieldname:ident;)*) => {$(
+
+        $crate::newtype_accessors_impls! {
+            $wrapper, $inner;
+        }
+
+        impl $wrapper {
+            #[doc="Size of the internal array"]
+            pub const SIZE: usize = $size;
+        }
+
+        impl AsRef<[u8]> for $wrapper {
+            fn as_ref(&self) -> &[u8] {
+                &(self.0).$fieldname[..]
+            }
+        }
+
+        impl AsMut<[u8]> for $wrapper {
+            fn as_mut(&mut self) -> &mut [u8] {
+                &mut (self.0).$fieldname[..]
+            }
+        }
+
+        impl<'bytes> TryFrom<&'bytes [u8]> for $wrapper {
+            type Error = $crate::FfiError;
+
+            fn try_from(src: &[u8]) -> ::core::result::Result<Self, Self::Error> {
+                if src.len() < $size {
```

Comment:
We should either use `Self::SIZE` or `$size`, but not both (note that there's also some usefulness to making `Self::SIZE` into a trait instead). Also, there's no reason to be lax in the size of the slice we accept, we're not returning anything useful to "advance" a buffer here.
```suggestion
                if src.len() != $size {
```

Reply:
While this is outside the scope of the PR now that this is unchanged,  I fixed it.

Reply:
On second thought, introducing this unrelated change breaks a few tests in the PR stack. Let's address this in a follow up PR since it isn't germane to the `Display` impl.